### PR TITLE
feat(erasing): add grayscale fill mode

### PR DESCRIPTION
**Pull request by federicopozzi33**
## Summary by Sourcery

Add a grayscale fill mode to erasing/cutout dropout transforms for images, volumes, and volume batches, with appropriate validation and support across the dropout pipeline.

New Features:
- Support using fill="grayscale" in Erasing and underlying cutout functions to convert erased regions to grayscale while preserving image shape and dtype for images, volumes, and batches.

Enhancements:
- Broaden BaseDropout and Erasing fill parameter types to accept a generic string mode and document the new grayscale behavior.
- Introduce helper functions to apply grayscale conversion to cutout holes for single images, single volumes, and batches of volumes, reusing common dropout infrastructure.
- Improve erasing parameter generation by returning an empty holes array directly when no valid region can be found.

Tests:
- Add tests covering grayscale fill behavior for single images, single volumes, and batches of volumes, including that only sampled regions are modified and that channels are equalized.
- Add validation tests ensuring grayscale fill is restricted to 1- or 3-channel inputs and that fill_mask must be None when using fill="grayscale".

### CODE REVIEW

File: albumentations/augmentations/dropout/functional.py [hunk 1]
```diff
@@ -158,6 +164,116 @@ def fill_holes_with_value(img: ImageType, holes: np.ndarray, fill: np.ndarray) -
...
+        ValueError: If image has channels other than 1 or 3.
+
+    """
+    num_channels = get_num_channels(img)
+
+    if holes.size == 0:
+        return img
+
+    if num_channels not in {1, 3}:
+        raise ValueError("Grayscale fill works only for 1 or 3 channel images")
```
Comment (ternaus) [inline albumentations/augmentations/dropout/functional.py]: There are many ways to convert to grayscale, most of them support any number of channels, no need to use `to_gray_weighted_average`

Comment (federicopozzi33) [reply albumentations/augmentations/dropout/functional.py]: Got it. I will address this comment in the next few days.

Comment (federicopozzi33) [reply albumentations/augmentations/dropout/functional.py]: @ternaus, fixed. Let me know if you'd like something a bit more sophisticated... for example, using the perceptual color conversion when RGB images are provided.
